File: frontend/html/scripts/canvas/canvasObjects/Points.js
import {CanvasObjectType} from "../CanvasObjectType.js";
import CanvasObject from "../CanvasObject.js";


export default class Points extends CanvasObject {
    constructor(reader) {
        super(CanvasObjectType.POINTS, reader);
        this.points = [];
        for (let i = 0; i < reader.readUint16(); i++) {
            this.points[i] = {
                dt:reader.readUint8(),
                x:reader.readUint16(),
                y:reader.readUint16()
            };
        }
    }

    draw(dt) {
        super.draw(dt);

    }

    serialize(writer) {
        super.serialize(writer);
    }
}

File: board-room/src/main/java/net/stzups/board/data/objects/canvas/objects/Points.java
package net.stzups.board.data.objects.canvas.objects;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.User;
import net.stzups.board.data.objects.canvas.CanvasObject;
import net.stzups.board.data.objects.canvas.CanvasObjectType;

public class Points extends CanvasObject {
    static class Point {
        public int dt;//todo byte type instead of int?
        public short x;
        public short y;

        public Point(int dt, short x, short y) {
            this.dt = dt;
            this.x = x;
            this.y = y;
        }
    }

    private Point[] points;

    public Points(ByteBuf byteBuf) {
        super(CanvasObjectType.POINTS, byteBuf);
        points = new Point[byteBuf.readShort()];
        for (int i = 0; i < points.length; i++) {
            points[i] = new Point(byteBuf.readByte(), byteBuf.readShort(), byteBuf.readShort());
        }
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeShort(points.length);
        for (Point point : points) {
            byteBuf.writeByte((byte) point.dt);
            byteBuf.writeShort(point.x);
            byteBuf.writeShort(point.y);
        }
    }
}
